### ISSUE TITLE
Fatal error postgresql_env

### ISSUE THREAD

**Issue by storeman** [ANXS/postgresql#88]
When running the postgresql role with only some modified postgresql.conf vars, I get the following error:

```
fatal: [db1]: FAILED! => {"msg": "ERROR! the field 'environment' has an invalid value (postgresql_env), and could not be converted to an dict. Error was: ", "failed": true}
```

I cannot really solve this.

**Comment by soupdiver** [comment]
Which fields you changed and what do you have for `postgresql_locale` variable?

**Comment by storeman** [comment]
Some fields for memory usage, really, only some postgresql.conf params. The `postgresql_locale` is unchanged. When I replaced postgresql_env with the content of it:

```
- name: PostgreSQL | Install PostgreSQL
  apt:
    name: "{{item}}"
    state: present
    update_cache: yes
    cache_valid_time: "{{apt_cache_valid_time | default (3600)}}"
  environment:
    LC_ALL: "{{ postgresql_locale }}"
    LC_LCTYPE: "{{ postgresql_locale }}"
    RUNLEVEL: 1
  with_items:
```

It works just fine.
    - "postgresql-{{postgresql_version}}"
    - "postgresql-client-{{postgresql_version}}"
    - "postgresql-contrib-{{postgresql_version}}"

**Comment by soupdiver** [comment]
I tested the playbook with the included `Vagrantfile` and it works fine.
Can you add a debug statement before the failing tasks and output the content of `postgresql_env `?

**Comment by storeman** [comment]
Ofcourse:

```
TASK [postgresql : postgresql : debug var=postgresql_env] ***********************
ok: [db1] => {
    "postgresql_env": {
        "LC_ALL": "en_US.UTF-8",
        "LC_LCTYPE": "en_US.UTF-8",
        "RUNLEVEL": 1
    },
    "changed": false
}
ok: [db2] => {
    "postgresql_env": {
        "LC_ALL": "en_US.UTF-8",
        "LC_LCTYPE": "en_US.UTF-8",
        "RUNLEVEL": 1
    },
    "changed": false
}

TASK [postgresql : postgresql : PostgreSQL | Install PostgreSQL] ****************
fatal: [db1]: FAILED! => {"msg": "ERROR! the field 'environment' has an invalid value (postgresql_env), and could not be converted to an dict. Error was: ", "failed": true}
```

**Comment by soupdiver** [comment]
maybe it is because of the `RUNLEVEL` can you try to set it to "1"?

**Comment by storeman** [comment]
This makes no difference. At first I didn't use RUNLEVEL, still the same error. I think this could be a python/ansible issue? I fully updated my Ubuntu version (14.04.2)

**Comment by soupdiver** [comment]
Which python version are you using on the machine?

**Comment by storeman** [comment]
Python 2.7.6

**Comment by soupdiver** [comment]
Ok that's weird... 
That's the same version I have running inside the VM.
I also tried adding `RUNLEVEL`to my env it it also worked.

So the problem seem somehow related to you installation but I'm not sure what exactly :/

@otakup0pe @farridav @pjan any ideas?

**Comment by storeman** [comment]
I had some other weird issues as well a few moments ago. When trying to get data from the hostvar[name] it gave an error with postgresql_admin_user. Even if I completely isolated the playbook. I figured there was something really wrong, but could nothing find about this UndefinedError. So I unstalled it via pip and removed all files and folders with ansible. After installing it again, this issue was resolved and the skipped messages are blue. Don't know what it was (grrr), but I'm glad this is fixed, as is my other issue with the hostvar. Thanks for your effort!
